src/hooks/things.ts

```diff
@@ -113,13 +112,15 @@ export function useResource(uri: SwrlitKey, options: SwrlitConfigInterface = {})
   const mutate = swrldResult.mutate
   const fetch = useFetcher(options.fetch)
   const saveResource = useCallback(async function (newDataset: SolidDataset) {
-    if (uri) {
-      mutate(newDataset, false)
-      const savedDataset = await saveSolidDatasetAt(uri, newDataset, { fetch })
-      mutate(savedDataset)
-      return savedDataset
+    if (uri && newDataset) {
```

Comment:
is this necessary? I guess we probably need it for extra error checking when working with plan JS, but the type of newDataset above doesn't allow for nulls, so in Typescript at least this check should be redundant?

Reply:
`saveResource` in `saveThing` above is potentially calling it with nil. I assume typescript doesn't know at runtime how to guard against that? Plus, I think our types right now for the callbacks are `any`, though I guess that doesn't matter too much for this since the type in in the signature.